### PR TITLE
Bland-Altman Plots for the Reliability Module

### PR DESCRIPTION
**Pull request by foger3**
fixes https://github.com/jasp-stats/jasp-issues/issues/494

### CODE REVIEW

**Comment by juliuspf** [comment]
Hi @foger3 thank you for making the plot available in JASP. Is there a particular reason why the plot is part of the intraclass correlation analysis? It is used for comparing two measurements, right? And usually the ICCs compare more than two measurement/ratings. Having said that, I am uncertain if it would better be within rater agreement, since it is not necessarily comparing raters but more generally measurements, and thus could confuse people. It could also get its own analysis tab underneath reliability, or we make a completely new ribbon for "agreement" given that there are more feature requests concerning that (https://github.com/jasp-stats/jasp-issues/issues/1755). Maybe @AlexanderLyNL or @vandenman have some input...

**Comment by Kucharssim** [comment]
Yes, we had a discussion with @foger3 about this a while ago, it indeed did not seem like this plot fits into any particular analysis... I am not sure what was the main reason to put it under a separate section in the ICC and not rater agreement, but this was definitely a factor:

> For the rater agreement analysis the data is read in as factors, which will not work with the plot

----

>My preference for now would be number 3, since the plot does not depend on any results from any of the other analyses.

Fair point, but making a whole new separate analysis just for one plot seems a bit odd, even though it does remove some of the awkwardness of using it within an analysis that does not really correspond to it. 

Another alternative would be to put it into descriptives? I am not sure whether it's good to make the descriptives more bloated than they already are for such a specialised plot, though.

> make a new ribbon "Agreement" distinct from "Reliability" and add the ICC, rater agreement, and Bland-Altman plot to it given that there are more feature requests concerning that

Not a bad idea, but not something we should do right now, I think... Once we support that one module can show as separate icons in the ribbon, we could do it that way (see discussion in https://github.com/jasp-stats/INTERNAL-jasp/issues/1812)

**Comment by Kucharssim** [comment]
> you can only use the variables of a data set in the plot when they have been added to the ICC analysis

I see your point. We could do by removing the `source` in the AvailableVariablesList component which will allow to use any variables, not only those that we compute the ICC for... However, yeah then it's kind of pointless to have it in the ICC in the first place. I think you changed my mind a little bit, perhaps it would be better to just make it a separate analysis...

**Comment by juliuspf** [comment]
I can move this into a separate analysis if that is okay with @foger3. I am not sure if I have permissions to make changes to this PR though... in the past this has often not worked.

**Review by Kucharssim — CHANGES_REQUESTED** [review]
Looks good, the only thing to do to finish this would be to remove this functionality from the ICC analysis, right? Otherwise the functionality is duplicated.

File: R/intraclassCorrelation.R [hunk 1]
```diff
@@ -22,6 +22,9 @@ intraclassCorrelation <- function(jaspResults, dataset, options) {
 
   jaspResults[["table"]] <- .handleIntraclassCorrelation(dataset, options)
 
+  .descriptivesBlandAltman(     jaspResults, dataset, options)
+  .descriptivesBlandAltmanTable(jaspResults, dataset, options)
```
Comment (Kucharssim) [inline R/intraclassCorrelation.R:26]: This also needs to go, right?

**Review by Kucharssim — CHANGES_REQUESTED** [review]

**Review by Kucharssim — APPROVED** [review]
Great! Thanks @juliuspf for taking over the PR!

**Pull request merged**